frontend/app/controllers/spree/checkout_controller.rb

```diff
@@ -52,6 +51,26 @@ def update
     end
 
     private
+
+      def update_params
+        if update_params = massaged_params[:order]
+          update_params.permit(permitted_checkout_attributes)
+        else
+          # We current allow update requests without any parameters in them.
+          {}
+        end
+      end
+
+      def massaged_params
+        massaged_params = params.deep_dup
+
+        move_payment_source_into_payments_attributes(massaged_params)
+        move_existing_card_into_payments_attributes(massaged_params)
```

Comment:
@mtomov that is because it is *not* deprecated here and is not intended to be, which actually goes along with my point that this is only coincidental (and temporary) duplication. :)